compiler/cpp/test/CMakeLists.txt

```diff
@@ -18,15 +18,15 @@
 #
 
 
-# Find required packages
-set(Boost_USE_STATIC_LIBS ON) # Force the use of static boost test framework
-find_package(Boost 1.53.0 REQUIRED COMPONENTS unit_test_framework)
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
+if(${WITH_PLUGIN})
+    # Find required packages
+    set(Boost_USE_STATIC_LIBS ON) # Force the use of static boost test framework
+    find_package(Boost 1.53.0 REQUIRED COMPONENTS unit_test_framework)
+    include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
 
-#Make sure gen-cpp files can be included
-include_directories("${CMAKE_CURRENT_BINARY_DIR}")
+    #Make sure gen-cpp files can be included
+    include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 
-if(${WITH_PLUGIN})
```

Comment:
Shared boost test support is added after this patch.
So we now need to fix the test.